src/App.js

```diff
@@ -1,13 +1,79 @@
 import React from 'react';
+import TodoList from './components/TodoComponents/TodoList';
+import TodoForm from './components/TodoComponents/TodoForm';
+import './App.css';
 
 class App extends React.Component {
   // you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor(props) {
+    super(props);
+    const savedTasksJson = window.localStorage.getItem('savedTasks');
+    this.state = { 
+      todosList: savedTasksJson ? JSON.parse(savedTasksJson) : [],
+      formField: "",
+    };
+  }
+
+  clickHandler = event => {
+    event.preventDefault();
+    if (event.target.id === 'addTaskButton'){
+      this.addTodo(this.state.formField);
+      this.setState({
+        formField: "",
+      });
+    } else if (event.target.id === 'clearCompletedButton') {
+      this.clearCompleted();
+    };
+  }
+
+  changeHandler = event => {
+    this.setState({
+      formField: event.target.value,
+    });
+  }
+
+  addTodo = (todoTask) => {
+    const newTodo = {
+      task: todoTask,
+      id: Date.now(),
```

Comment:
Id as a date is fine here, but I believe it's something you should try to avoid, because of the real possibility of ids with equal values. It's probably better to use a library like uuid for this.